src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -146,6 +148,19 @@ internal sealed class ClassEvaluator : MemberEvaluator {
             }
         }
 
+        /// <summary>
+        /// Decides if this class is an abstract class or not
+        /// </summary>
+        private void DecideAbstract() {
+            // A class is derived from an abstract class if one of its bases is abstract or one of its bases is also derived from an abstract class
+            var derivedFromAbstract = _class.Bases.OfType<PythonClassType>().Any(b => b.IsAbstract || b.IsDerivedFromAbstractClass);
+            // A class is abstract if it is derived from an abstract class and has at least one abstract function
+            var isAbstract = derivedFromAbstract && _class.GetMembers<PythonFunctionType>().Any(f => f.IsAbstract);
```

Comment:
`IPythonFunctionType`. Are there abstract properties? https://docs.python.org/3/library/abc.html#abc.abstractproperty

Reply:
The link you gave says abstractproperties are deprecated. But for Python 2.7 I believe we should handle this.